File: VB/SchedulerAPISample/CodeExamples/Serialization.vb
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Xml
Imports System

Namespace SchedulerAPISample.CodeExamples
    Friend Class SerializationExamples
        Private Shared Sub AppointmentSerialization(ByVal scheduler As SchedulerControl)
'            #Region "#AppointmentSerialization"
            SchedulerCompatibility.Base64XmlObjectSerialization = True
            For Each apt As Appointment In scheduler.ActiveView.GetAppointments()
                Dim helper As New AppointmentXmlPersistenceHelper(apt, Nothing)
                apt.Description = helper.ToXml()
            Next apt
'            #End Region ' #AppointmentSerialization
        End Sub

        Private Shared Sub AppointmentDeserialization(ByVal scheduler As SchedulerControl)
'            #Region "#AppointmentDeserialization"
            scheduler.Storage.Appointments.Clear()
            Dim xmlAppointment As String = "<Appointment Version=""1"" Start=""01/13/2016 04:22:00"" " & ControlChars.CrLf & _
"End=""01/13/2016 05:53:00"" Label=""5"" ResourceId=""3"" Status=""2"" Subject=""Meet Andrew Miller"" />"
            Dim objAppointment As Appointment = AppointmentXmlPersistenceHelper.ObjectFromXml(scheduler.Storage.Appointments, xmlAppointment)
            scheduler.Storage.Appointments.Add(objAppointment)
            scheduler.Start = New Date(2016, 1, 13)
'            #End Region ' #AppointmentDeserialization
        End Sub
    End Class
End Namespace

File: CS/SchedulerAPISample/CodeExamples/Serialization.cs
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Xml;
using System;

namespace SchedulerAPISample.CodeExamples {
    class SerializationExamples {
        static void AppointmentSerialization(SchedulerControl scheduler) {
            #region #AppointmentSerialization
            SchedulerCompatibility.Base64XmlObjectSerialization = true;
            foreach (Appointment apt in scheduler.ActiveView.GetAppointments()) {
                AppointmentXmlPersistenceHelper helper = 
                    new AppointmentXmlPersistenceHelper(apt, null);
                apt.Description = helper.ToXml();
            }
            #endregion #AppointmentSerialization
        }

        static void AppointmentDeserialization(SchedulerControl scheduler) {
            #region #AppointmentDeserialization
            scheduler.DataStorage.Appointments.Clear();
            string xmlAppointment = @"<Appointment Version=""1"" Start=""01/13/2016 04:22:00"" 
End=""01/13/2016 05:53:00"" Label=""5"" ResourceId=""3"" Status=""2"" Subject=""Meet Andrew Miller"" />";
            Appointment objAppointment = AppointmentXmlPersistenceHelper.ObjectFromXml(scheduler.DataStorage.Appointments, xmlAppointment);
            scheduler.DataStorage.Appointments.Add(objAppointment);
            scheduler.Start = new DateTime(2016, 1, 13);
            #endregion #AppointmentDeserialization
        }
    }
}
